tests/testthat/test-dotdotdot.R

```diff
@@ -0,0 +1,56 @@
+test_that("unpack_list", {
+  # unpacks a single list
+  expect_identical(
+    unpack_list(list(1, 2, 3)),
+    unpack_list(1, 2, 3)
+  )
+
+  # ... not if more args
+  expect_failure(expect_equal(
+    unpack_list(list(1, 2, 3), "a"),
+    unpack_list(1, 2, 3, "a")
+  ))
+
+  # skip_classes
+  expect_identical(
+    unpack_list(data.frame(a = 1, b = 2), skip_classes = "data.frame"),
+    list(data.frame(a = 1, b = 2))
+  )
+  expect_failure(expect_equal(
+    unpack_list(data.frame(a = 1, b = 2), skip_classes = NULL),
+    list(data.frame(a = 1, b = 2))
+  ))
+
+  # trailing commas not allowed
+  err = unpack_list(1, .context = "this test", .call = call("foobar"), ) |>
+    result() |>
+    unwrap_err()
+  expect_identical(
+    err$contexts()[[1]],
+    "trailing argument commas are not (yet) supported with polars"
+  )
+  expect_identical(err$get_rcall(), "foobar()")
+  expect_identical(err$get_rinfo(), "this test")
+})
+
+
+
+test_that("unpack_bool_expr_result", {
+  # unpacks a single list, return ok-result, get same polars expression
+  a = unpack_bool_expr_result(pl$col("a") == 1, pl$col("b") > 2)
+  b = unpack_bool_expr_result(list(pl$col("a") == 1, pl$col("b") > 2))
+  expect_true(a$ok$meta$eq(b$ok))
+
+  # ... not if more args
+  expect_failure(expect_equal(
+    unpack_list(list(1, 2, 3), "a"),
+    unpack_list(1, 2, 3, "a")
+  ))
```

Comment:
Ditto.

Reply:
Normally I don't unittest internal functions too much. @etiennebacher  asked for it and since this one is very central I wrote the tests

I think this is in fact a very normal situation and a central  expected behavior 


```R
# inputs to DataFrame a list and a vector of same length
> pl$DataFrame(list(1,2,3),1:3)
shape: (3, 2)
┌────────────┬──────────────┐
│ new_column ┆ new_column_1 │
│ ---        ┆ ---          │
│ list[f64]  ┆ i32          │
╞════════════╪══════════════╡
│ [1.0]      ┆ 1            │
│ [2.0]      ┆ 2            │
│ [3.0]      ┆ 3            │
└────────────┴──────────────┘

# in the single case where input is a list and no other args, the list is unpacked
> pl$DataFrame(list(1,2,3))
shape: (1, 3)
┌────────────┬──────────────┬──────────────┐
│ new_column ┆ new_column_1 ┆ new_column_2 │
│ ---        ┆ ---          ┆ ---          │
│ f64        ┆ f64          ┆ f64          │
╞════════════╪══════════════╪══════════════╡
│ 1.0        ┆ 2.0          ┆ 3.0          │
└────────────┴──────────────┴──────────────┘

```

Reply:
in py-polars dicts are "unpacked" to columns. We allow both `data.frame` like interface where each column is an argument but also the py-polars way where all columns can be passed as a single named list.

```py
In [9]: pl.DataFrame({"a": [1,2,3], "b": [1,2,3]})
Out[9]: 
shape: (3, 2)
┌─────┬─────┐
│ a   ┆ b   │
│ --- ┆ --- │
│ i64 ┆ i64 │
╞═════╪═════╡
│ 1   ┆ 1   │
│ 2   ┆ 2   │
│ 3   ┆ 3   │
└─────┴─────┘
```

Reply:
What I would like to point out is that `expect_*` functions are nested

Reply:
ohh misunderstood what you meant, what you prefer instead?

tried this 
<img width="327" alt="image" src="https://github.com/pola-rs/r-polars/assets/8015271/541d78ff-662f-43f6-a89b-83a9989cd95d">

it seem the counting of number of tests is correct
<img width="437" alt="image" src="https://github.com/pola-rs/r-polars/assets/8015271/c124e5cb-de4c-42bf-b071-a8fcbb01f710">


I guess expect_success will catch the fail_test exception, such that testthat only sees the outer exception, if any.

Reply:
Thanks for the clarification. There doesn't seem to be a problem with this usage

Reply:
> Normally I don't unittest internal functions. @etiennebacher asked for it 

Just to clarify I wanted to test the new error messages from `$select()` and others, not necessarily test the internal functions directly. I'm fine with this.